prog/size.go

```diff
@@ -106,3 +106,45 @@ func (target *Target) assignSizesArray(args []Arg) {
 func (target *Target) assignSizesCall(c *Call) {
 	target.assignSizesArray(c.Args)
 }
+
+func (r *randGen) mutateSize(arg *ConstArg, parent []Arg) {
+	// Small sizes are generated when we mutate size target and update sizes,
+	// so here we mostly do large sizes that potentially cause overflows.
+	if r.oneOf(100) {
+		arg.Val = r.rand64()
+		return
+	}
+	typ := arg.Type().(*LenType)
+	elemSize := typ.ByteSize
+	if elemSize == 0 {
+		for _, field := range parent {
+			if typ.Buf == field.Type().FieldName() {
+				if arrayType, ok := field.Type().(*ArrayType); ok && !arrayType.Varlen() {
+					elemSize = arrayType.Type.Size()
+				}
+				break
+			}
+		}
+		if elemSize == 0 {
+			elemSize = 1
+		}
+	}
```

Comment:
When calculating `elemSize`:
1. You're missing a case when field has `VmaType`.
2. `field` can be a pointer, and in that case length denotes the length of the pointee object.
3.  Length field can point to the parent or one of the parent structures.

I think it makes sense to separate `getElemSize` method from `assignSizes` in `prog/size.go` and reuse here.

Reply:
1, 2 done
do we need to do anything special for 3? in that case elemSize is always 1, because that's not an array

there does not seem to be much to share between this function and assignSizes